### PR TITLE
fix(ModelComponents): Added functions to create Components with a giv…

### PR DESCRIPTION
**Pull request by MelchiorSchuh**
…en uuid.

### CODE REVIEW

File: include/geode/model/representation/builder/detail/copy.h [hunk 1]
```diff
@@ -65,6 +65,37 @@ namespace geode
...
+        template < typename ModelFrom, typename BuilderTo >
+        void copy_corner_components(
+            const ModelFrom& from, BuilderTo& builder_to, Mapping& mapping )
+        {
+            for( const auto& corner : from.corners() )
+            {
+                if( mapping.has_mapping_input( corner.id() ) )
+                {
+                    const auto& id = mapping.in2out( corner.id() );
+                    if( ModelFrom::dim == BuilderTo::dim )
```
Comment (BotellaA) [inline include/geode/model/representation/builder/detail/copy.h:77]: Using different syntax here and in the else is confusing

File: include/geode/model/representation/builder/detail/copy.h [hunk 2]
```diff
@@ -117,6 +211,37 @@ namespace geode
...
+                        builder_to.add_block( id );
+                    }
+                    builder_to.set_block_name( id, block.name() );
+                }
+                else
+                {
+                    const auto& id =
+                        ModelFrom::dim == BuilderTo::dim
+                            ? builder_to.add_block( block.mesh().impl_name() )
+                            : builder_to.add_block();
```
Comment (panquez) [inline include/geode/model/representation/builder/detail/copy.h:238]: is impl_name dependant of the dimension?

**Review by BotellaA — APPROVED** [review]

File: include/geode/model/representation/core/mapping.h [hunk 3]
```diff
@@ -71,6 +71,11 @@ namespace geode
             mappings.erase( type );
         }
 
+        MappingType& operator[]( const ComponentType& type )
+        {
+            return mappings[type];
+        }
+
```
[the remark this answers is not in the document]
Comment (MelchiorSchuh) [reply include/geode/model/representation/core/mapping.h:78]: But [] will create the element if it does not exist in the map, which is what I want

File: include/geode/model/representation/builder/detail/copy.h (hunk 2 above)
Comment (MelchiorSchuh) [reply include/geode/model/representation/builder/detail/copy.h:238]: Yes, impl_name will return the MeshImpl, which is dependant on the dimension (ex: OpenGeodePointSet2D)

**Pull request merged**